src/main/java/org/snomed/snowstorm/core/data/services/MultiSearchService.java

```diff
@@ -75,23 +63,39 @@ public class MultiSearchService implements CommitListener {
 	
 	private final Logger logger = LoggerFactory.getLogger(getClass());
 	
-	Map<String, String> publishedBranches = new HashMap<>();
-	
-	BoolQueryBuilder cachedBranchesQuery = null;
-	LocalDate cacheDate = null;
+	private final Map<String, String> publishedBranches = new HashMap<>();
 
-	public Page<Description> findDescriptions(DescriptionCriteria criteria, PageRequest pageRequest) {
-		
-		SearchHits<Description> searchHits = findDescriptionsHelper(criteria, pageRequest);
+	private MultiBranchCriteria cachedBranchCriteria = null;
+	private LocalDate cacheDate = null;
+
+	public Page<Description> findDescriptions(DescriptionCriteria criteria, String ecl, PageRequest pageRequest, Set<String> includeBranches) {
+
+		MultiBranchCriteria branchesQuery = new MultiBranchCriteria("all-released-plus-requested",getBranchesQuery().getTimepoint(),new ArrayList<>(getBranchesQuery().getBranchCriteria()));
+		if (includeBranches != null) {
+			for(String branch : includeBranches) {
+				branchesQuery.getBranchCriteria().add(versionControlHelper.getBranchCriteria(branch));
+			}
+		}
```

Comment:
@RichardAWood, sending a payload with empty branches does return a 400 Bad Request, but the request needs to reach ElasticSearch before it fails:

```
{
    "branches": [
        " "
    ]
}
```

Reply:
That's a good point.  Which do you think would be better - skipping blank branches:

			for(String branch : includeBranches) {
				if(branch != null && !branch.isBlank()) {
					branchesQuery.getBranchCriteria().add(versionControlHelper.getBranchCriteria(branch));
				}
			}

Or considering it a bad call, and throwing an error?

			for(String branch : includeBranches) {
				if(branch == null || branch.isBlank()) {
					throw new IllegalStateException("Cannot pass in an empty branch.");
				}
				else {
					branchesQuery.getBranchCriteria().add(versionControlHelper.getBranchCriteria(branch));
				}
			}

Reply:
@dmcgihtsdo  ElasticVC does the check here. https://github.com/IHTSDO/elasticvc/blob/master/src/main/java/io/kaicode/elasticvc/api/MultiBranchCriteria.java#L36

Reply:
Ah, great - I will not make any changes here, then.